### PR TITLE
Setup basic CI based on scylla_rust_driver

### PR DESCRIPTION
**Pull request by wmitros**

### CODE REVIEW

**Comment by wmitros** [comment]
Looks like the newest clippy is not passing on scylla-cql, this shouldn't be a problem when we make it just a dependency and not include its entire implementation here

**Comment by wmitros** [comment]
I saw that the new lint from clippy conflicts with rust-analyzer (https://github.com/scylladb/scylla-rust-driver/commit/4964b4471c3d14bfea75e532b2e7c24cbd48cd65), so I disabled it as well

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - name: Test
+      run: CARGO_TARGET_WASM32_WASI_RUNNER="/home/runner/.wasmtime/bin/wasmtime --invoke _start --allow-unknown-exports" cargo test --target=wasm32-wasi --all-targets
+
+  # Tests that our current minimum supported rust version compiles everything sucessfully
+  min_rust:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install Rust ${{ env.rust_min }}
+      uses: dtolnay/rust-toolchain@master
```
Comment (piodul) [inline .github/workflows/rust.yml:45]: Why did you choose to use this repo instead of `actions-rs/toolchain`?

**Review by piodul — APPROVED** [review]
Just one question, but looks good otherwise.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (wmitros) [reply .github/workflows/rust.yml:45]: I think I was getting some warnings about deprecated Node.js 12 actions, which were caused by the fact that the `actions-rs/toolchain` hasn't been maintained for over a year

**Pull request merged**